src/anomalib/utils/metrics/thresholding/adaptive_score_threshold.py

```diff
@@ -11,16 +11,14 @@
 from torch import Tensor
 from torchmetrics import PrecisionRecallCurve
 
+from .base import BaseAnomalyScoreThreshold
 
-class AnomalyScoreThreshold(PrecisionRecallCurve):
-    """Anomaly Score Threshold.
 
+class AdaptiveScoreThreshold(PrecisionRecallCurve, BaseAnomalyScoreThreshold):
```

Comment:
Is this method also called as `AdaptiveScoreThreshold` in the Mvtec paper?
What if we had another method that adaptively computes the threshold?

Reply:
They don't have this but yeah the naming is quite broad. How about F1AdaptiveThreshold?

Reply:
@djdameln, would be good to get your thoughts here too

Reply:
I would prefer `F1Adaptivethreshold`

Reply:
Alright let's go with `F1Adaptivethreshold` then